internal/dbx/measurement_test.go

```diff
@@ -0,0 +1,338 @@
+		t.Fail()
+	}
+	var expectedAmount int64 = 7 * 8 / 2
+	if ms3.AmountMicrocredits.Int64 != expectedAmount {
+		t.Logf("expected measurement 3 AmountMicrocredits to be %v, got %v", expectedAmount, ms3.AmountMicrocredits.Int64)
+	}
+	ms4 := ms[measurementFromReadingID(ms, readingID4)]
+	if !ms4.AmountMicrocredits.Valid {
+		t.Logf("expected measurement 4 AmountMicrocredits to be valid, but was not")
+		t.Fail()
+	}
+	if ms4.AmountMicrocredits.Int64 != expectedAmount {
+		t.Logf("expected measurement 4 AmountMicrocredits to be %v, got %v", expectedAmount, ms4.AmountMicrocredits.Int64)
+	}
+	ms5 := ms[measurementFromReadingID(ms, readingID5)]
+	if !ms5.AmountMicrocredits.Valid {
+		t.Logf("expected measurement 5 AmountMicrocredits to be valid, but was not")
+		t.Fail()
+	}
+	if ms5.AmountMicrocredits.Int64 != expectedAmount {
+		t.Logf("expected measurement 5 AmountMicrocredits to be %v, got %v", expectedAmount, ms5.AmountMicrocredits.Int64)
+	}
+	ms6 := ms[measurementFromReadingID(ms, readingID6)]
+	if ms6.AmountMicrocredits.Valid {
+		t.Logf("expected measurement 6 AmountMicrocredits to be invalid, but was valid")
+		t.Fail()
+	}
+}
+
+func measurementFromReadingID(m []db.Measurement, id int32) int {
+	return slices.IndexFunc(m, func(e db.Measurement) bool {
+		return e.ReadingID == id
+	})
+}
+
+func TestDBBoundsMonthPrev(t *testing.T) {
+	tz, _ := time.LoadLocation("America/New_York")
+
+	testCases := []struct {
+		Name                string
+		Tz                  *time.Location
+		AsOf                pgtype.Timestamptz
+		ExpectedPeriodStart pgtype.Timestamptz
+		ExpectedPeriodEnd   pgtype.Timestamptz
+	}{
+		{
+			Name:                "AsOf on exclusive upper bound",
+			Tz:                  tz,
+			AsOf:                testutil.NewPgxTimestamptz(time.Date(2025, time.February, 1, 0, 0, 0, 0, tz)),
+			ExpectedPeriodStart: testutil.NewPgxTimestamptz(time.Date(2025, time.January, 1, 0, 0, 0, 0, tz)),
+			ExpectedPeriodEnd:   testutil.NewPgxTimestamptz(time.Date(2025, time.February, 1, 0, 0, 0, 0, tz)),
+		},
+		{
+			Name:                "AsOf mid-month",
+			Tz:                  tz,
+			AsOf:                testutil.NewPgxTimestamptz(time.Date(2025, time.February, 15, 0, 0, 0, 0, tz)),
+			ExpectedPeriodStart: testutil.NewPgxTimestamptz(time.Date(2025, time.January, 1, 0, 0, 0, 0, tz)),
+			ExpectedPeriodEnd:   testutil.NewPgxTimestamptz(time.Date(2025, time.February, 1, 0, 0, 0, 0, tz)),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			// Arrange
+			conn, err := pgxpool.New(t.Context(), "")
+			if err != nil {
+				t.Fatal("creating database connection failed", err)
+			}
+			// Test acquiring a connection from the pool.
+			if err = conn.Ping(t.Context()); err != nil {
+				t.Fatal("database connection ping failed")
+			}
+			q := dbx.NewQuerier(db.New(conn))
+
+			// Act
+			result, err := q.BoundsMonthPrev(t.Context(), tc.AsOf)
```

Comment:
suggestion: missing error handler here
```suggestion
			if result, err := q.BoundsMonthPrev(t.Context(), tc.AsOf); err != nil {
				t.Fatalf("error occured getting bounds: %v", err)
			}
```